test/nfs_test.rb

```diff
@@ -281,22 +311,156 @@ def find_mp(path)
       expect(find_mp("/foo").passno).to eq 0
       expect(find_mp("/foo/bar/baz").passno).to eq 0
     end
+
+    shared_examples "remove_entry" do
+      it "removes the existing NFS share" do
+        nfs_sid = nfs(spec).sid
+
+        subject.WriteOnly
+
+        expect(working_graph.find_device(nfs_sid)).to be_nil
+      end
+    end
+
+    context "when the entry is new" do
+      before do
+        entry = entry(spec)
+
+        if entry
+          entry["new"] = true
+        else
+          entry = create_entry(spec)
+          entry["new"] = true
+          subject.nfs_entries << entry
+        end
+      end
+
+      shared_examples "new_entry" do
+        it "creates a new NFS share for that entry" do
+          subject.WriteOnly
+
+          nfs = nfs(spec)
+
+          expect(nfs).to_not be_nil
+          expect(nfs.exists_in_probed?).to eq(false)
+        end
+
+        it "sets the NFS mount point as active" do
+          subject.WriteOnly
+
+          nfs = nfs(spec)
+
+          expect(nfs.mount_point.active?).to eq(true)
+        end
+
+        it "sets the NFS to be written to the fstab" do
+          subject.WriteOnly
+
+          nfs = nfs(spec)
+
+          expect(nfs.mount_point.in_etc_fstab?).to eq(true)
+        end
+      end
+
+      context "and the NFS share does no exist in the system yet" do
+        let(:spec) { "nfs.example.com:/home/test" }
+
+        include_examples "new_entry"
+      end
+
+      context "and the NFS share already exists in the system" do
+        let(:spec) { "nfs.example.com:/foo" }
+
+        before do
+          # Set mount point properties of the existing NFS share. This configuration should not be
+          # propagated to the new NFS share.
+          nfs = nfs(spec)
+          nfs.mount_point.active = false
+          nfs.mount_point.in_etc_fstab = false
+        end
+
+        include_examples "remove_entry"
+
+        include_examples "new_entry"
+      end
+    end
+
+    context "when the entry is not new" do
+      before do
+        entry = entry(spec)
+        entry["new"] = false
+      end
+
+      let(:spec) { "nfs.example.com:/foo" }
+
+      include_examples "remove_entry"
+
+      context "and the entry corresponds to a mounted NFS share" do
+        before do
+          nfs(spec).mount_point.active = true
+        end
+
+        it "creates a NFS share with active mount point" do
+          subject.WriteOnly
+
+          expect(nfs(spec).mount_point.active?).to eq(true)
+        end
+      end
+
+      context "and the entry corresponds to an unmounted NFS share" do
+        before do
+          nfs(spec).mount_point.active = false
+        end
+
+        it "creates a NFS share with inactive mount point" do
+          subject.WriteOnly
+
+          expect(nfs(spec).mount_point.active?).to eq(true)
```

Comment:
This expectation seems wrong, accordding to the subject

> creates a NFS share with **inactive** mount point

Am I missing something?

Reply:
Yes, it is wrong. Thanks!